File: examples/python/accept.py
"""Check work another agent handed back before you pay for it or pass it on.

    pip install -r requirements.txt
    CROSSCHECK_WALLET_KEY=0x... python accept.py task.txt deliverable.txt
"""

import asyncio
import sys

from pay import post_and_wait


async def main(task_file: str, deliverable_file: str) -> int:
    out = await post_and_wait(
        "https://crosscheckapi.com/v1/accept",
        {"task": open(task_file, encoding="utf-8").read(), "deliverable": open(deliverable_file, encoding="utf-8").read()},
    )
    v = out["verdict"]
    print("ACCEPT" if v["accept"] else "REJECT", "-", v["summary"])
    for r in v["requirements"]:
        print(f"  {'x' if r['blocking'] else 'ok'} {r['met']:<11} {r['requirement']}: {r['evidence']}")
    print("proof:", out.get("share_url"))
    return 0 if v["accept"] else 1


if __name__ == "__main__":
    sys.exit(asyncio.run(main(sys.argv[1], sys.argv[2])))

File: examples/typescript/accept.ts
// Check work another agent handed back before you pay for it or pass it on.
//
//   CROSSCHECK_WALLET_KEY=0x... npx tsx accept.ts task.txt deliverable.txt
import { readFileSync } from "node:fs";
import { postAndWait } from "./pay.js";

const [taskFile, deliverableFile] = process.argv.slice(2);
if (!taskFile || !deliverableFile) throw new Error("Usage: accept.ts <task-file> <deliverable-file>");
const out = await postAndWait("https://crosscheckapi.com/v1/accept", {
  task: readFileSync(taskFile, "utf8"),
  deliverable: readFileSync(deliverableFile, "utf8"),
});
const v = out.verdict;
console.log(v.accept ? "ACCEPT" : "REJECT", "-", v.summary);
for (const r of v.requirements) console.log(`  ${r.blocking ? "x" : "ok"} ${r.met.padEnd(11)} ${r.requirement}: ${r.evidence}`);
console.log("proof:", out.share_url);
process.exitCode = v.accept ? 0 : 1;
